index.d.ts

```diff
@@ -0,0 +1,304 @@
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<DeleteOutput<Schema>>;
+    put(
+      item: Schema,
+      options?: {
+        conditions?: [] | {};
+        capacity?: NoneOption | TotalOption | IndexesOption; // Cannot use DynamoDB type as it's upper-case
+        metrics?: NoneOption | SizeOption;
+        returnValues?: NoneOption | AllOldOption;
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<PutOutput<Schema>>;
+    update(
+      key: Schema & { $remove?: string[] }, // Complex part to type
+      options?: {
+        conditions?: [] | {};
+        capacity?: NoneOption | TotalOption | IndexesOption;
+        metrics?: NoneOption | SizeOption;
+        returnValues?:
+          | NoneOption
+          | AllOldOption
+          | UpdatedOldOption
+          | AllNewOption
+          | UpdatedNewOption;
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<UpdateOutput<Schema>>;
+    query(
+      partitionKey: string,
+      options?: QueryInputOptions,
+      parameters?
+    ): Promise<any>;
+    scan(options?, parameters?);
+  }
+
+  // Table
+  interface TableAttributes {
+    [attribute: string]: AnyDynamoDBType;
+  }
+  interface Indexes {
+    [index: string]: { partitionKey?: string; sortKey?: string };
+  }
+  interface TableConstructor {
+    name: string;
+    alias?: string;
+    partitionKey: string;
+    sortKey?: string;
+    entityField?: boolean | string;
+    attributes?: TableAttributes;
+    indexes?: Indexes;
+    autoExecute?: boolean;
+    autoParse?: boolean;
+    DocumentClient?: DocumentClient;
+    entities?: {}; // improve - not documented
+  }
+
+  // Entity
+  interface EntityAttributeConfiguration<Schema> {
+    type?: AnyDynamoDBType;
+    coerce?: boolean;
+    default?: SchemaType | ((data: Schema) => SchemaType); // Value "type" or a function
+    onUpdate?: boolean;
+    required?: boolean | AlwaysOption;
+    alias?: string;
+    map?: string;
+    setType?: DynamoDBStringType | DynamoDBNumberType | DynamoDBBinaryType;
+    partitionKey?: boolean | string;
+    sortKey?: boolean | string;
+  }
+
+  interface VisibleEntityAttributeConfiguration<Schema>
+    extends EntityAttributeConfiguration<Schema> {
+    hidden?: false;
+  }
+
+  interface HiddenEntityAttributeConfiguration<Schema>
+    extends EntityAttributeConfiguration<Schema> {
+    hidden: true;
+  }
+
+  type CompositeKeyConfiguration<Schema, EAC> = EAC & {
+    type?: DynamoDBStringType | DynamoDBNumberType | DynamoDBBooleanType;
+    save?: boolean;
+  };
+
+  type CompositeKey<Schema, EAC> = [
+    string,
+    number,
+    (
+      | CompositeKeyConfiguration<Schema, EAC>
+      | DynamoDBStringType
+      | DynamoDBNumberType
+      | DynamoDBBooleanType
+    )?
+  ];
+
+  type EntityAttributes<Schema, HiddenKeys extends Partial<string>> = Record<
+    keyof Schema,
+    AnyDynamoDBType | any
```

Comment:
Hi @xavierlefevre, thanks for your work on this.

it could be like that, right?
```suggestion
    AnyDynamoDBType | EntityAttributeConfiguration<Schema>
```